classes/abstracts/ActionScheduler_Abstract_QueueRunner.php

```diff
@@ -79,6 +82,43 @@ public function process_action( $action_id, $context = '' ) {
 		}
 	}
 
+	/**
+	 * Determine if action should be retried and requeue if necessary.
+	 *
+	 * The retry logic requires that the action is not recurring, and that the $retry array exists and is not empty. A
+	 * `failures` key keeps track of Action fails. `count` limits the number of retries (i.e. $failures <= $count),
+	 * and can be overridden by filter. Default is 5.
+	 *
+	 * Future work: add key/variables for different backoff algorithms.
+	 *
+	 * @param ActionScheduler_Action $action The action to evaluate.
+	 */
+	protected function retry_action( $action ) {
+		// Don't retry recurring actions.
+		if ( ! isset( $action ) || ! is_a( $action, 'ActionScheduler_Action' ) || $action->get_schedule()->is_recurring() ) {
+			return;
+		}
+		// Return when $retry is empty.
+		$retry = $action->get_retry();
+		if ( empty( $retry ) || ! $retry->is_valid_after_fail() ) {
+			return;
+		}
+
+		// Default backoff algorithm is exponential (2^n * 100).  The backoff value will be added to time() to
+		// schedule when the webhook should be retried (in seconds).  If an invalid value (not numeric) is returned
+		// DON'T schedule the retry.
+		$backoff = pow( 2, $retry->get_fails() ) * 100;
+		$backoff = apply_filters( 'action_scheduler_retry_backoff', $backoff, $action );
+		if ( empty( $backoff ) || ! is_numeric( $backoff ) ) {
+			return;
+		}
+
+		$date = as_get_datetime_object( time() + abs( (int) $backoff ) );
+		$schedule = new ActionScheduler_SimpleSchedule( $date );
```

Comment:
this is duplicate implementation of creating an action. lets just use `ActionScheduler::factory()->single( ` here instead